controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -633,11 +633,20 @@ private Scope getScope(final String scopeName) {
         return getScaleMetadata(scopeName, streamName, null, executor).thenApply(scaleMetadataList -> {
             long size = scaleMetadataList.size();
             long totalNumSplits = 0;
-
-            for (int i = 0, j = 1; j < size; j++) {
-                long countSplitsEventI = scaleMetadataList.get(i).getSegments().size();
-                long countSplitsEventJ = scaleMetadataList.get(j).getSegments().size();
-                totalNumSplits = (countSplitsEventI < countSplitsEventJ) ? (countSplitsEventJ - countSplitsEventI) : 0;
+            long countSplitsEventI, countSplitsEventI1;
+
+            if (isDescendingOrder(scaleMetadataList)) {
```

Comment:
as discussed, we dont need if block here..
you can compare timestamps of scaleMetadataList.get(i).getTimestamp and scaleMetadataList.get(i+1).getTimestamp and based on that figure out direction and compute splits and merges. 
you dont need isdescending check